### PR TITLE
Treat * as input solvable specially

### PR DESCRIPTION
**Pull request by lubomir**
If the input specifies * as a solvable, it means the user wants to include all packages in the repo. No masking should be done. However if a package is provided in a module as well as a bare RPM, the best solvable function will keep only one version.
    
The fix for this is to ignore all modular packages when a wildcard is processed. The modular packages will still be included because their module will be included. Only latest version of the bare RPM will get in.

This PR includes commits from #47 as well, since those touch the same code.

### CODE REVIEW

File: fus.c [hunk 1]
```diff
@@ -867,16 +867,61 @@ resolve_all_solvables (Pool  *pool,
...
+ */
+static void
+remove_from_selection (Pool  *pool,
+                       Queue *selection,
+                       Id     solvid)
+{
+    g_auto(Queue) remove;
+    queue_init (&remove);
+    selection_make (pool, &remove, pool_solvid2str (pool, solvid), SELECTION_CANON);
+    selection_subtract (pool, selection, &remove);
```
Comment (r4f4) [inline fus.c:884]: You can replace this `selection_subtract` call by a `SELECTION_SUBTRACT` flag in `selection_make`. You don't even have to create the Queue.

Comment (r4f4) [reply fus.c:884]: Which means this function is not even needed.

**Comment by r4f4** [comment]
Stupid question: is this a valid case for depsolving? If you want everything in the repo, there is no solving to do...

**Comment by lubomir** [comment]
The validity is complicated. Unfortunately from compose point of view everything is not actually everything. It means all native packages; and not all multilib. I'll have a look if it's possible to expand the list outside of fus and present list of NVRs so that this hack is not needed.

**Comment by lubomir** [comment]
I can expand the wildcard to exactly the NEVRAs that should be included. The remaining problem is that fus will reject some of them since they are masked by a package in some module. Maybe we could update the masking to only apply if the package is selected by name only.

**Comment by lubomir** [comment]
Changing the filtering is actually really easy. I added it to #47. Please take a look at the latest commit there.

**Pull request closed without merging**